ndcube/utils/collection.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import astropy.units as u
+
+
+def _sanitize_user_aligned_axes(data, aligned_axes):
+    """
+    Converts input aligned_axes to standard format.
+    aligned_axes can be supplied by the user in a few ways:
+    *. A tuple of tuples of ints, where each tuple corresponds to a cube
+    in the collection, and each int designates the an aligned axis in numpy order.
+    In this case, the axis represented by the 0th int in the 0th tuple is aligned
+    with the 0th int in the 1st tuple and so on.
+    *. A single tuple of ints if all aligned axes are in the same order.
+    *. A single int if only one axis is aligned and if the aligned axis in each cube
+    is in the same order.
+    """
+    aligned_axes_error_message = "aligned_axes must contain ints or " + \
+        "a tuple of ints for each element in data."
+    if isinstance(data[0].dimensions, u.Quantity):
+        cube0_dims = data[0].dimensions[np.array(aligned_axes[0])]
+    elif isinstance(data[0].dimensions, tuple):
+        cube0_dims = np.array(data[0].dimensions, dtype=object)[np.array(aligned_axes[0])]
+    else:
+        raise TypeError("0th cube in collection of unsupported type: {0}".format(type(data[0])))
```

Comment:
What exactly is this doing, do you need the different behaviour on type here, they look kinda similar?

Reply:
One is for the ```NDCube``` case the other is for the ```NDCubeSequence``` case.  Remember that
```type(NDCube.dimensions)``` is ```u.Quantity``` while ```type(NDCubeSequence.dimensions)``` is a ```tuple``` of ```u.Quantity```.

However, I could change this to handle the ```NCubeSequence``` case as special and assume all other cases are compatible with the ```NDCube``` one.